mava/systems/tf/maddpg/execution.py

```diff
@@ -367,9 +367,32 @@ def select_actions(
 
         actions = {}
         policies = {}
+        new_states = {}
         for agent, observation in observations.items():
-            actions[agent], policies[agent] = self.select_action(agent, observation)
-        return actions, policies
+            actions[agent], policies[agent], new_states[agent] = self.select_action(
+                agent, observation, states[agent]
+            )
+        return actions, policies, new_states
+
+    def select_action(  # type: ignore
```

Comment:
Why do we have the type ignore here? Seems like basic typing.

Reply:
I think it was not conforming to the recurrent executor's argument setup. We now use tf.function which means the agent_state should also be passed in as an argument. We could maybe update the recurrent executor definition when all the systems use tf.function over the select_actions function.

Reply:
Ok thanks, perhaps we can make an issue for this.

Reply:
Will do.